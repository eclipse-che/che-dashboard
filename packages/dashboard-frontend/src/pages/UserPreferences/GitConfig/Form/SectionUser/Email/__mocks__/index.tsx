/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import * as React from 'react';

import { Props } from '..';

export class GitConfigUserEmail extends React.PureComponent<Props> {
  public render(): React.ReactElement {
    const { onChange, isLoading, value } = this.props;

    return (
      <div data-testid="config-user-email">
        <span data-testid="isLoading">{isLoading}</span>
        <span data-testid="value">{value}</span>
        <button onClick={() => onChange('new-user@che', true)}>Change Email Valid</button>
        <button onClick={() => onChange('new-user@che', false)}>Change Email Invalid</button>
      </div>
    );
  }
}
