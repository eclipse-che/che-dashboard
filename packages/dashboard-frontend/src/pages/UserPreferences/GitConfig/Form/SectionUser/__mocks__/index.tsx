/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import * as React from 'react';

import { Props } from '..';

export class GitConfigSectionUser extends React.PureComponent<Props> {
  render(): React.ReactElement {
    const { config, onChange } = this.props;
    return (
      <div>
        <div data-testid="user-email">{config.user.email}</div>
        <div data-testid="user-name">{config.user.name}</div>
        <button
          onClick={() => onChange({ user: { email: 'new-user@che', name: 'new user' } }, true)}
        >
          Change Email Valid
        </button>
      </div>
    );
  }
}
