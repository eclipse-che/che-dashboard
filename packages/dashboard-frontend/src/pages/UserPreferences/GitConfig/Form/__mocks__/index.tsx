/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '..';

export class GitConfigForm extends React.PureComponent<Props> {
  render(): React.ReactElement {
    const { isLoading, gitConfig, onReload, onSave } = this.props;

    return (
      <div data-testid="git-config-form">
        <span data-testid="is-loading">{isLoading}</span>
        <span data-testid="git-config">{JSON.stringify(gitConfig)}</span>
        <button onClick={() => onSave({ user: { email: 'new-user@che', name: 'new user name' } })}>
          Save Config
        </button>
        <button onClick={() => onReload()}>Reload Config</button>
      </div>
    );
  }
}
