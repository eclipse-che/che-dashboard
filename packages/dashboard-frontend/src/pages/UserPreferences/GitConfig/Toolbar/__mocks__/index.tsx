/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitConfig/Toolbar';

export class GitConfigToolbar extends React.PureComponent<Props> {
  render() {
    const { mode, onAdd, onChangeMode } = this.props;
    const nextMode = mode === 'form' ? 'viewer' : 'form';
    return (
      <div>
        Mock Git Config Toolbar
        <span data-testid="toolbar-mode">{mode}</span>
        <button data-testid="toolbar-on-add" onClick={onAdd}>
          Add
        </button>
        <button data-testid="toolbar-on-change-mode" onClick={() => onChangeMode(nextMode)}>
          Switch Mode
        </button>
      </div>
    );
  }
}
