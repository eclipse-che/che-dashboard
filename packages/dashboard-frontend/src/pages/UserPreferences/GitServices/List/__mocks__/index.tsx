/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitServices/List';

export class GitServicesList extends React.PureComponent<Props> {
  render(): React.ReactNode {
    const { isDisabled, gitOauth, providersWithToken, skipOauthProviders, onRevokeServices } =
      this.props;

    return (
      <div data-testid="git-services-list">
        <div>GitServicesList</div>
        <div data-testid="number-of-git-services">{gitOauth.length}</div>
        <div data-testid="providers-with-token">{providersWithToken.join()}</div>
        <div data-testid="skip-oauth-providers">{skipOauthProviders.join()}</div>
        <div data-testid="list-is-disabled">{isDisabled ? 'true' : 'false'}</div>
        <button onClick={() => onRevokeServices(gitOauth)}>Revoke</button>
      </div>
    );
  }
}
