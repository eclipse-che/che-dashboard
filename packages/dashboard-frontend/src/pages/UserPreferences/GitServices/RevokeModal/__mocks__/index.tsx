/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitServices/RevokeModal';

export class GitServicesRevokeModal extends React.PureComponent<Props> {
  render() {
    const { isOpen, onCancel, onRevoke } = this.props;
    return (
      <div data-testid="git-services-revoke-modal">
        <div>GitServicesRevokeModal</div>
        <div data-testid="revoke-modal-is-open">{isOpen ? 'open' : 'closed'}</div>
        <button onClick={onRevoke}>Revoke</button>
        <button onClick={onCancel}>Cancel</button>
      </div>
    );
  }
}
