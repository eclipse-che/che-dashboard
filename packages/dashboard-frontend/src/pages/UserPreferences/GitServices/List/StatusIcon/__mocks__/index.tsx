/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitServices/List/StatusIcon';

export class GitServiceStatusIcon extends React.PureComponent<Props> {
  public render(): React.ReactElement {
    const { gitProvider } = this.props;

    return (
      <span data-testid="git-service-status-icon">
        <span>Authentication Status Icon for {gitProvider}</span>
      </span>
    );
  }
}
