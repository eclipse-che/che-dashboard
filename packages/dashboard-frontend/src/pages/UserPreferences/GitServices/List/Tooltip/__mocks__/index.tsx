/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitServices/List/Tooltip';

export class GitServiceTooltip extends React.PureComponent<Props> {
  public render(): React.ReactElement {
    const { isVisible } = this.props;
    return (
      <span data-testid="git-service-tooltip">
        <span>isVisible: {isVisible ? 'true' : 'false'}</span>
      </span>
    );
  }
}
