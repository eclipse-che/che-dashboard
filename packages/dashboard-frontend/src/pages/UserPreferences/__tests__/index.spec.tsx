/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import userEvent from '@testing-library/user-event';
import React from 'react';
import { Provider } from 'react-redux';
import { Location } from 'react-router-dom';

import UserPreferences from '@/pages/UserPreferences';
import getComponentRenderer, { screen } from '@/services/__mocks__/getComponentRenderer';
import { buildUserPreferencesLocation } from '@/services/helpers/location';
import { UserPreferencesTab } from '@/services/helpers/types';
import { FakeStoreBuilder } from '@/store/__mocks__/storeBuilder';

jest.mock('../ContainerRegistriesTab');
jest.mock('../GitConfig');
jest.mock('../GitServices');
jest.mock('../PersonalAccessTokens');
jest.mock('../SshKeys');

const { renderComponent } = getComponentRenderer(getComponent);

const mockNavigate = jest.fn();

function getComponent(location: Location): React.ReactElement {
  const store = new FakeStoreBuilder().build();
  return (
    <Provider store={store}>
      <UserPreferences location={location} navigate={mockNavigate} />
    </Provider>
  );
}

describe('UserPreferences', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('snapshot', () => {
    const location = buildUserPreferencesLocation();
    renderComponent(location);

    expect(document.body).toMatchSnapshot();
  });

  it('should activate the Container Registries tab by default', () => {
    const location = buildUserPreferencesLocation('unknown-tab-name' as UserPreferencesTab);

    renderComponent(location);

    expect(screen.queryByRole('tabpanel', { name: 'Container Registries' })).toBeTruthy();
  });

  describe('Location change', () => {
    it('should activate the Container Registries tab', () => {
      const location = buildUserPreferencesLocation(UserPreferencesTab.CONTAINER_REGISTRIES);
      renderComponent(location);

      expect(screen.queryByRole('tabpanel', { name: 'Container Registries' })).toBeTruthy();
    });

    it('should activate the Git Services tab', () => {
      const location = buildUserPreferencesLocation(UserPreferencesTab.GIT_SERVICES);
      renderComponent(location);

      expect(screen.queryByRole('tabpanel', { name: 'Git Services' })).toBeTruthy();
    });

    it('should activate the Personal Access Tokens tab', () => {
      const location = buildUserPreferencesLocation(UserPreferencesTab.PERSONAL_ACCESS_TOKENS);
      renderComponent(location);

      expect(screen.queryByRole('tabpanel', { name: 'Personal Access Tokens' })).toBeTruthy();
    });

    it('should activate the SSH Keys tab', () => {
      const location = buildUserPreferencesLocation(UserPreferencesTab.SSH_KEYS);
      renderComponent(location);

      expect(screen.queryByRole('tabpanel', { name: 'SSH Keys' })).toBeTruthy();
    });
  });

  describe('Tabs', () => {
    it('should activate the Container Registries tab', async () => {
      const location = buildUserPreferencesLocation(UserPreferencesTab.SSH_KEYS);
      renderComponent(location);

      const tab = screen.getByRole('tab', { name: 'Container Registries' });
      await userEvent.click(tab);

      expect(screen.queryByRole('tabpanel', { name: 'Container Registries' })).toBeTruthy();
    });

    it('should activate the Git Services tab', async () => {
      const location = buildUserPreferencesLocation(UserPreferencesTab.SSH_KEYS);
      renderComponent(location);

      const tab = screen.getByRole('tab', { name: 'Git Services' });
      await userEvent.click(tab);

      expect(screen.queryByRole('tabpanel', { name: 'Git Services' })).toBeTruthy();
    });

    it('should activate the Personal Access Tokens tab', async () => {
      const location = buildUserPreferencesLocation(UserPreferencesTab.SSH_KEYS);
      renderComponent(location);

      const tab = screen.getByRole('tab', { name: 'Personal Access Tokens' });
      await userEvent.click(tab);

      expect(screen.queryByRole('tabpanel', { name: 'Personal Access Tokens' })).toBeTruthy();
    });

    it('should activate the Gitconfig tab', async () => {
      const location = buildUserPreferencesLocation(UserPreferencesTab.SSH_KEYS);
      renderComponent(location);

      const tab = screen.getByRole('tab', { name: 'Gitconfig' });
      await userEvent.click(tab);

      expect(screen.queryByRole('tabpanel', { name: 'Gitconfig' })).toBeTruthy();
    });

    it('should activate the SSH Keys tab', async () => {
      const location = buildUserPreferencesLocation(UserPreferencesTab.CONTAINER_REGISTRIES);
      renderComponent(location);

      const tab = screen.getByRole('tab', { name: 'SSH Keys' });
      await userEvent.click(tab);

      expect(screen.queryByRole('tabpanel', { name: 'SSH Keys' })).toBeTruthy();
    });
  });
});
