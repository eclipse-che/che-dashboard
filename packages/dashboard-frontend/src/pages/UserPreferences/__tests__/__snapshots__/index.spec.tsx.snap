// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Preferences should correctly render the component which contains two registries 1`] = `
Array [
  <section
    className="pf-c-page__main-section pf-m-light"
  >
    <h1
      className="pf-c-title pf-m-2xl"
    >
      User Preferences
    </h1>
  </section>,
  <div
    className="pf-c-tabs"
    data-ouia-component-id="OUIA-Generated-Tabs-4"
    data-ouia-component-type="PF4/Tabs"
    data-ouia-safe={true}
    id="user-preferences-tabs"
    onSelect={[Function]}
    style={
      Object {
        "backgroundColor": "var(--pf-global--BackgroundColor--100)",
      }
    }
  >
    <button
      aria-hidden={false}
      aria-label="Scroll left"
      className="pf-c-tabs__scroll-button"
      disabled={false}
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        aria-labelledby={null}
        fill="currentColor"
        height="1em"
        role="img"
        style={
          Object {
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
    </button>
    <ul
      className="pf-c-tabs__list"
      onScroll={[Function]}
    >
      <li
        className="pf-c-tabs__item pf-m-current"
      >
        <button
          aria-controls="pf-tab-section-container-registries-container-registries-tab"
          className="pf-c-tabs__link"
          data-ouia-component-type="PF4/TabButton"
          data-ouia-safe={true}
          id="pf-tab-container-registries-container-registries-tab"
          onClick={[Function]}
        >
          Container Registries
        </button>
      </li>
    </ul>
    <button
      aria-hidden={false}
      aria-label="Scroll right"
      className="pf-c-tabs__scroll-button"
      disabled={false}
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        aria-labelledby={null}
        fill="currentColor"
        height="1em"
        role="img"
        style={
          Object {
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        />
      </svg>
    </button>
  </div>,
  <section
    aria-labelledby="pf-tab-container-registries-container-registries-tab"
    className="pf-c-tab-content"
    data-ouia-component-type="PF4/TabContent"
    data-ouia-safe={true}
    hidden={false}
    id="pf-tab-section-container-registries-container-registries-tab"
    role="tabpanel"
    tabIndex={0}
  >
    <span>
      
    </span>
    <section
      className="pf-c-page__main-section"
    >
      <div
        className="pf-c-toolbar pf-m-page-insets"
        id="registries-list-toolbar"
      >
        <div
          className="pf-c-toolbar__content"
        >
          <div
            className="pf-c-toolbar__content-section"
          >
            <div
              className="pf-c-toolbar__item"
            >
              <button
                aria-disabled={true}
                aria-label={null}
                className="pf-c-button pf-m-danger pf-m-disabled"
                data-ouia-component-id="OUIA-Generated-Button-danger-2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                data-testid="bulk-delete-button"
                disabled={true}
                onClick={[Function]}
                role={null}
                tabIndex={null}
                type="button"
              >
                Delete
              </button>
            </div>
            <div
              className="pf-c-toolbar__item pf-m-align-right-on-md pf-m-align-right-on-lg pf-m-align-right-on-xl pf-m-align-right-on-2xl"
            >
              <button
                aria-disabled={false}
                aria-label={null}
                className="pf-c-button pf-m-link"
                data-ouia-component-id="OUIA-Generated-Button-link-4"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                data-testid="add-button"
                disabled={false}
                onClick={[Function]}
                role={null}
                type="button"
              >
                <span
                  className="pf-c-button__icon pf-m-start"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                      transform=""
                    />
                  </svg>
                </span>
                Add Container Registry
              </button>
            </div>
          </div>
          <div
            className="pf-c-toolbar__expandable-content"
            id="registries-list-toolbar-expandable-content-1"
          >
            <div
              className="pf-c-toolbar__group"
            />
          </div>
        </div>
        <div
          className="pf-c-toolbar__content pf-m-hidden"
          hidden={true}
        >
          <div
            className="pf-c-toolbar__group"
          />
        </div>
      </div>
      <table
        aria-label="Container credentials"
        className="pf-c-table pf-m-grid-md pf-m-compact"
        data-ouia-component-id={5}
        data-ouia-component-type="PF4/Table"
        data-ouia-safe={true}
        role="grid"
      >
        <thead
          className=""
        >
          <tr>
            <td
              className="pf-c-table__check"
              data-key={0}
              data-label=""
              onMouseEnter={[Function]}
            >
              <input
                aria-label="Select all rows"
                checked={false}
                name="check-all"
                onChange={[Function]}
                type="checkbox"
              />
              
            </td>
            <th
              className=""
              data-key={1}
              data-label="Host"
              onMouseEnter={[Function]}
              scope="col"
            >
              Host
            </th>
            <th
              className=""
              data-key={2}
              data-label="Username"
              onMouseEnter={[Function]}
              scope="col"
            >
              Username
            </th>
            <td
              className=""
              data-key={3}
              data-label=""
              onMouseEnter={[Function]}
            >
              
            </td>
          </tr>
        </thead>
        <tbody
          className=""
        >
          <tr
            className=""
            data-ouia-component-id={6}
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe={true}
            hidden={false}
            onMouseDown={[Function]}
          >
            <td
              className="pf-c-table__check"
              data-key={0}
              onMouseEnter={[Function]}
            >
              <input
                aria-labelledby="simple-node0"
                checked={false}
                name="checkrow0"
                onChange={[Function]}
                type="checkbox"
              />
            </td>
            <td
              className=""
              data-key={1}
              data-label="Host"
              onMouseEnter={[Function]}
            >
              <a
                href="http://test.reg"
                rel="noreferrer"
                target="_blank"
              >
                http://test.reg
              </a>
            </td>
            <td
              className=""
              data-key={2}
              data-label="Username"
              onMouseEnter={[Function]}
            />
            <td
              className="pf-c-table__action"
              data-key={3}
              onMouseEnter={[Function]}
            >
              <div
                className="pf-c-dropdown pf-m-align-right"
                data-ouia-component-id={7}
                data-ouia-component-type="PF4/Dropdown"
                data-ouia-safe={true}
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-label="Actions"
                  className="pf-c-dropdown__toggle pf-m-plain"
                  disabled={false}
                  id="pf-toggle-id-2"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 192 512"
                    width="1em"
                  >
                    <path
                      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                      transform=""
                    />
                  </svg>
                </button>
              </div>
            </td>
          </tr>
          <tr
            className=""
            data-ouia-component-id={8}
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe={true}
            hidden={false}
            onMouseDown={[Function]}
          >
            <td
              className="pf-c-table__check"
              data-key={0}
              onMouseEnter={[Function]}
            >
              <input
                aria-labelledby="simple-node1"
                checked={false}
                name="checkrow1"
                onChange={[Function]}
                type="checkbox"
              />
            </td>
            <td
              className=""
              data-key={1}
              data-label="Host"
              onMouseEnter={[Function]}
            >
              <a
                href="https://tstreg.com"
                rel="noreferrer"
                target="_blank"
              >
                https://tstreg.com
              </a>
            </td>
            <td
              className=""
              data-key={2}
              data-label="Username"
              onMouseEnter={[Function]}
            />
            <td
              className="pf-c-table__action"
              data-key={3}
              onMouseEnter={[Function]}
            >
              <div
                className="pf-c-dropdown pf-m-align-right"
                data-ouia-component-id={9}
                data-ouia-component-type="PF4/Dropdown"
                data-ouia-safe={true}
              >
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-label="Actions"
                  className="pf-c-dropdown__toggle pf-m-plain"
                  disabled={false}
                  id="pf-toggle-id-3"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 192 512"
                    width="1em"
                  >
                    <path
                      d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                      transform=""
                    />
                  </svg>
                </button>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </section>
  </section>,
]
`;

exports[`User Preferences should correctly render the component without registries 1`] = `
Array [
  <section
    className="pf-c-page__main-section pf-m-light"
  >
    <h1
      className="pf-c-title pf-m-2xl"
    >
      User Preferences
    </h1>
  </section>,
  <div
    className="pf-c-tabs"
    data-ouia-component-id="OUIA-Generated-Tabs-2"
    data-ouia-component-type="PF4/Tabs"
    data-ouia-safe={true}
    id="user-preferences-tabs"
    onSelect={[Function]}
    style={
      Object {
        "backgroundColor": "var(--pf-global--BackgroundColor--100)",
      }
    }
  >
    <button
      aria-hidden={false}
      aria-label="Scroll left"
      className="pf-c-tabs__scroll-button"
      disabled={false}
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        aria-labelledby={null}
        fill="currentColor"
        height="1em"
        role="img"
        style={
          Object {
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
        />
      </svg>
    </button>
    <ul
      className="pf-c-tabs__list"
      onScroll={[Function]}
    >
      <li
        className="pf-c-tabs__item pf-m-current"
      >
        <button
          aria-controls="pf-tab-section-container-registries-container-registries-tab"
          className="pf-c-tabs__link"
          data-ouia-component-type="PF4/TabButton"
          data-ouia-safe={true}
          id="pf-tab-container-registries-container-registries-tab"
          onClick={[Function]}
        >
          Container Registries
        </button>
      </li>
    </ul>
    <button
      aria-hidden={false}
      aria-label="Scroll right"
      className="pf-c-tabs__scroll-button"
      disabled={false}
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        aria-labelledby={null}
        fill="currentColor"
        height="1em"
        role="img"
        style={
          Object {
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 256 512"
        width="1em"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        />
      </svg>
    </button>
  </div>,
  <section
    aria-labelledby="pf-tab-container-registries-container-registries-tab"
    className="pf-c-tab-content"
    data-ouia-component-type="PF4/TabContent"
    data-ouia-safe={true}
    hidden={false}
    id="pf-tab-section-container-registries-container-registries-tab"
    role="tabpanel"
    tabIndex={0}
  >
    <span>
      
    </span>
    <section
      className="pf-c-page__main-section"
    >
      <div
        className="pf-c-empty-state pf-m-sm pf-m-full-height"
      >
        <div
          className="pf-c-empty-state__content"
        >
          <svg
            aria-hidden="true"
            aria-labelledby={null}
            className="pf-c-empty-state__icon"
            fill="currentColor"
            height="1em"
            role="img"
            style={
              Object {
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 64 1204 1024"
            width="1em"
          >
            <path
              d="M799.314 529.6c3.2 5.257 4.8 11.2 4.8 17.371v226.971c0 7.086-2.057 13.714-5.943 19.657-4.114 5.943-9.371 10.057-16 12.571l-184.914 68.571c-3.886 1.371-7.771 2.057-11.657 2.057-0.229 0-0.457 0-0.457 0-0.229 0-0.457 0-0.457 0-3.886 0-7.771-0.686-11.657-2.057l-184.914-68.571c-6.629-2.514-11.886-6.629-16-12.571s-5.943-12.343-5.943-19.657v-226.971c0-6.171 1.6-12.114 4.8-17.371s7.543-9.6 12.8-12.571l184.914-73.829c4.8-2.743 10.286-4.343 16-4.343 0.229 0 0.457 0 0.457 0s0.457 0 0.457 0c5.714 0 11.2 1.371 16 4.343l184.914 73.829c5.257 2.971 9.6 7.086 12.8 12.571zM770.286 574.628l-185.371-82.286-185.6 82.286v32.229l185.6-78.4 185.371 78.4v-32.229zM770.286 678.857l-185.143-84.8-185.6 84.8v35.886l185.6-78.171 185.371 79.314-0.229-37.029zM486.4 355.2l-216 80c-4.571 1.6-8.914 2.514-13.486 2.514-0.229 0-0.457 0-0.686 0s-0.457 0-0.686 0c-4.571 0-8.914-0.914-13.486-2.514l-216-80c-7.771-2.971-13.943-7.771-18.743-14.629-5.029-6.857-7.314-14.629-7.314-22.857v-264.686c0-7.314 1.829-14.171 5.486-20.343s8.686-11.2 15.086-14.629l216-86.171c5.714-3.429 11.886-5.029 18.743-5.029 0.229 0 0.457 0 0.686 0s0.457 0 0.686 0c6.857 0 13.029 1.6 18.743 5.029l216 110.4c6.4 3.429 11.429 8.457 15.086 14.629s5.486 13.029 5.486 20.343v240.457c0 8.229-2.286 16-7.086 22.857-4.571 6.857-10.743 11.657-18.514 14.629zM472.686 87.543l-216.686-98.514-216.914 96v37.714l216.914-91.429 216.686 93.943v-37.714zM472.686 206.857l-216.686-98.971-216.914 98.971v41.829l216.914-91.2 216.686 92.571v-43.2zM1144.457 355.2l-216 80c-4.571 1.6-8.914 2.514-13.486 2.514-0.229 0-0.457 0-0.686 0s-0.457 0-0.686 0c-4.571 0-8.914-0.914-13.486-2.514l-216-80c-7.771-2.971-13.943-7.771-18.743-14.629s-7.086-14.4-7.086-22.857v-264.686c0-7.314 1.829-14.171 5.486-20.343s8.686-11.2 15.086-14.629l216-86.171c5.714-3.429 11.886-5.029 18.743-5.029 0.229 0 0.457 0 0.686 0s0.457 0 0.686 0c6.857 0 13.029 1.6 18.743 5.029l216 110.4c6.4 3.429 11.429 8.457 15.086 14.629s5.486 13.029 5.486 20.343v240.457c0 8.229-2.286 16-7.086 22.857s-10.971 11.657-18.743 14.629zM1130.971 87.543l-216.686-98.514-216.914 96v37.714l216.914-91.429 216.686 93.943v-37.714zM1130.971 206.857l-216.686-98.971-216.914 98.971v41.829l216.914-91.2 216.686 92.571v-43.2z"
              transform="rotate(180 0 512) scale(-1 1)"
            />
          </svg>
          <h4
            className="pf-c-title pf-m-lg"
          >
            No Container Registries
          </h4>
          <div
            className="pf-c-empty-state__body"
          >
            <button
              aria-disabled={false}
              aria-label="add-registry"
              className="pf-c-button pf-m-link"
              data-ouia-component-id="OUIA-Generated-Button-link-2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe={true}
              disabled={false}
              onClick={[Function]}
              role={null}
              type="button"
            >
              <span
                className="pf-c-button__icon pf-m-start"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                    transform=""
                  />
                </svg>
              </span>
              Add Container Registry
            </button>
          </div>
        </div>
      </div>
    </section>
  </section>,
]
`;
